Bump version, reset build number for jsoncpp v1.8.4

chambbj opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

chambbj committed:
Bump version, reset build number for jsoncpp v1.8.4

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

chambbj commented:
@conda-forge-admin, please rerender

conda-forge-admin committed:
MNT: Re-rendered with conda-smithy 3.1.5 and pinning 2018.05.22

chambbj committed:
Fix Windows paths for tests

chambbj committed:
Merge branch 'jsoncpp-1.8.4' of https://github.com/chambbj/jsoncpp-feedstock into jsoncpp-1.8.4

chambbj committed:
Lib name has changed

jakirkham commented:
~~This may no longer compile with VS 2008. Would recommend skipping that build and re-rendering.~~

NVM found an upstream patch that fixes this issue.

xref: https://github.com/open-source-parsers/jsoncpp/pull/695#issuecomment-353173044

jakirkham committed:
Include upstreamed patch for VS2008 support

Upstream made a change that broke pre-C++11 support, which means VS 2008
failed to build. This includes a patch from upstream that should fix
this issue.

jakirkham commented on recipe/meta.yaml:
```diff
@@ -7,10 +7,17 @@ package:
+    #                                                                 #
+    #  xref: https://github.com/open-source-parsers/jsoncpp/pull/733  #
+    ###################################################################
+    - PR_733.patch
```
xref: https://github.com/open-source-parsers/jsoncpp/pull/733

chambbj commented:
Thanks for locating the missing piece @jakirkham. LGTM

jakirkham commented:
@conda-forge-admin, please re-render.

conda-forge-admin committed:
MNT: Re-rendered with conda-smithy 3.1.6 and pinning 2018.06.19

jakirkham commented:
Strange, re-rendering seems to have added a `python` matrix even though `python` is in `requirements/build` and not in `requirements/host`, which also exists. Any ideas?

cc @isuruf

jakirkham committed:
Revert "MNT: Re-rendered with conda-smithy 3.1.6 and pinning 2018.06.19"

This reverts commit 8d0295b29b28d5e39fadcd9881e74059683c0352.

jakirkham committed:
Add explicit, empty `requirements/run`

Seems to help avoid adding `python` into the CI matrix locally.

jakirkham commented:
@conda-forge-admin, please re-render.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I rerendered the feedstock and it seems to be already up-to-date.

jakirkham commented:
Seems adding an empty, explicit `requirements/run` section addressed the issue. Though it doesn't seem like this should have mattered.

jakirkham commented:
Raised issue ( https://github.com/conda/conda-build/issues/2989 ) upstream about the `python` matrix that showed up.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham commented:
Thanks @chambbj :)
